### ISSUE TITLE
macos: use "Main Thread Checker" sanitizer?

### ISSUE THREAD

**Issue by bnoordhuis** [libuv/libuv#4188]
From https://developer.apple.com/documentation/xcode/diagnosing-memory-thread-and-crash-issues-early:

> The Main Thread Checker tool dynamically replaces system methods that must execute on the main thread with variants that check the current thread. The tool replaces only system APIs with well-known thread requirements, and doesn’t replace all system APIs. Because the replacements occur in system frameworks, Main Thread Checker doesn’t require you to recompile your app.

And also:

> Because Main Thread Checker doesn’t require you to recompile your code, you can run it on an existing macOS binary. Inject the dynamic library located at /Applications/Xcode.app/Contents/Developer/usr/lib/libMainThreadChecker.dylib into your executable.

**Comment by juanarbol** [comment]
I could work on this one, I just need a bit of time for it

**Comment by bnoordhuis** [comment]
Investigating if - and if so, where - it makes sense to use is part of the task.

**Comment by juanarbol** [comment]
I'm currently investigating. For now, besides 200 warnings about types and losing precision, and I don't see a lot of value

**Comment by juanarbol** [comment]
Honestly. I don't see anything useful from this. I think -based on 10 mins reading- that is mostly for UI updates in SWIFT

![Screenshot 2024-05-21 at 23 14 45](https://github.com/libuv/libuv/assets/17013303/031e384c-ead9-4113-ba13-018a5625412f)

**Comment by juanarbol** [comment]
Yes, I think it is for updates that must be called from the main thread.

https://developer.apple.com/documentation/xcode/diagnosing-memory-thread-and-crash-issues-early#Detect-improper-UI-updates-on-background-threads

>The Main Thread Checker tool ensures that all calls that must occur on the main thread do so. To enable this tool, select Main Thread Checker from the Diagnostics section of the appropriate scheme.

**Comment by juanarbol** [comment]
As the "Main Thread Checker" is not a tool for libuv, should this be closed?

**Comment by bnoordhuis** [comment]
If that's the conclusion, then I'll go ahead and close this.
